### ISSUE TITLE
mtd.restore() crashes with ChildFailedError after training completes successfully

### ISSUE THREAD

**Issue by TheSabari07** [NVIDIA/Model-Optimizer#2486]
## Describe the bug

I noticed an issue when calling `mtd.restore(distillation_model)` after a knowledge distillation training run has completed successfully.

The training itself completes normally, all steps finish, and checkpoints are saved to disk. However, when I call `mtd.restore(distillation_model)` as part of the cleanup/save step, it crashes with a `ChildFailedError` (exit code 1 from `torch.distributed`).

I may be misunderstanding the intended usage of `mtd.restore()` here, so I wanted to check whether this behavior is expected.

**Impact:** The training completes successfully and the checkpoints can still be recovered using a manual workaround, but this seems to prevent using the intended clean restore/save flow.

### Steps/Code to reproduce bug

1. Wrap a Hugging Face causal LM student model:

```python
distillation_model = mtd.convert(
    student_model,
    mode=[("kd_loss", {
        "teacher_model": teacher_model,
        "criterion": mtd.LogitsDistillationLoss(temperature=2.0),
        "loss_balancer": mtd.StaticLossBalancer({
            "kd_loss": 0.7,
            "ce_loss": 0.3,
        }),
    })]
)
```

2. Train to completion using `transformers.Trainer`, launched with:

```bash
torchrun --nproc_per_node=1 train_decoder.py
```

3. After `trainer.train()` returns successfully and checkpoints exist on disk, call:

```python
mtd.restore(distillation_model)
```

4. The crash occurs at this line.

### Expected behavior

I expected `mtd.restore()` to cleanly detach the distillation wrapper and return the plain student model so that it can then be saved using `save_pretrained()`.

Could someone please clarify whether `mtd.restore()` is expected to be called after `Trainer.train()` in this setup, or if there is another recommended way to restore the student model before saving?

## System information

* Container used (if applicable): none / bare venv
* OS: Ubuntu Linux
* CPU architecture: x86_64
* GPU name: NVIDIA DGX B200
* GPU memory size: 192GB
* Number of GPUs: 1 (via `CUDA_VISIBLE_DEVICES`, single-GPU run)
* Library versions:

  * Python: 3.12.3
  * ModelOpt version or commit hash: 0.46.1
  * CUDA: 13.2.86
  * PyTorch: 2.12.0.dev20260408+cu128
  * Transformers: 5.14.1
  * TensorRT-LLM: N/A
  * ONNXRuntime: N/A
  * TensorRT: N/A
* Any other details: Teacher = Qwen2.5-Coder-7B-Instruct, Student = Qwen2.5-Coder-0.5B-Instruct, launched via `torchrun --nproc_per_node=1` (not multi-GPU DDP).
